assets/src/models/createDetourMachine.ts

```diff
@@ -348,51 +358,35 @@ export const createDetourMachine = setup({
               },
             },
             "Finished Drawing": {
-              invoke: [
-                {
-                  src: "fetch-finished-detour",
-                  input: ({
-                    context: { routePattern, startPoint, endPoint },
-                  }) => ({
-                    routePatternId: routePattern?.id,
-                    startPoint,
-                    endPoint,
-                  }),
-
-                  onDone: {
-                    actions: assign({
-                      finishedDetour: ({ event }) => event.output,
-                    }),
-                  },
-
-                  onError: {},
-                },
-
-                {
-                  src: "fetch-detour-directions",
-                  input: ({
-                    context: { startPoint, waypoints, endPoint },
-                  }) => ({
-                    points: (startPoint ? [startPoint] : [])
-                      .concat(waypoints || [])
-                      .concat(endPoint ? [endPoint] : []),
-                  }),
+              invoke: {
+                src: "fetch-finished-detour",
+                input: ({
+                  context: { routePattern, startPoint, waypoints, endPoint },
+                }) => ({
+                  routePatternId: routePattern?.id,
+                  startPoint,
+                  waypoints,
+                  endPoint,
+                }),
 
-                  onDone: {
-                    actions: assign({
-                      detourShape: ({ event: { output: detourShape } }) =>
-                        map(detourShape, (shape) => ({
-                          ...shape,
-                          directions: shape.directions?.concat({
+                onDone: {
+                  actions: assign({
+                    finishedDetour: ({ event }) => event.output,
+                    detourShape: ({ event }) =>
+                      event.output?.detourShape &&
+                      Ok({
+                        ...event.output.detourShape,
+                        directions: event.output.detourShape.directions?.concat(
+                          {
                             instruction: "Regular Route",
-                          }),
-                        })),
-                    }),
-                  },
-
-                  onError: {},
+                          }
+                        ),
```

Comment:
note: I do wonder if we should be surfacing the potential errors from directions, so I've asked in Slack https://mbta.slack.com/archives/C024N3SKX33/p1720010049121199

Reply:
This PR does mean that there's one interaction that used to cause a visible error banner that would now not do that. I don't love the idea of introducing a bug, but in the grand scheme of things, this feels relatively low-impact.

Here's why I thought this was okay:
- There are two user interactions that can create this error banner: placing a waypoint and placing the end point.
- If you'd get the error from placing a waypoint, you'll still get that error.
- So the only actual change is if an error happens that would only manifest in the directions between the last waypoint and the end point, at which point you get the bad UX of no feedback and no route shape.
- I'm guessing ☝️ is super unlikely, if not impossible, which is why I don't feel strongly that we need to fix this right this instant.

The only reason that I'm sort of okay with moving forward with introducing the bug is that it does unblock the MQTT shape work (you can't build a `Shape` without info from the consolidated `/finished_detour` endpoint), and I don't know what the Right™ thing to do here is.